app/models/dialog_field_drop_down_list.rb

```diff
@@ -3,7 +3,51 @@ def show_refresh_button?
     !!show_refresh_button
   end
 
+  def force_multi_value
+    return true if options[:force_multi_value].present? &&
+                   options[:force_multi_value] != "null" &&
+                   options[:force_multi_value]
+  end
+
+  def force_multi_value=(setting)
+    options[:force_multi_value] = setting
+  end
+
   def initial_values
     [[nil, "<None>"]]
   end
+
+  def refresh_json_value(checked_value)
+    self.default_value = nil
+    @raw_values = nil
+
+    refreshed_values = values
+
+    selectbox_options = refreshed_values.collect { |value_pair| value_pair[0].to_s }
+
+    @value = if checked_value.kind_of?(Array) && (selectbox_options & checked_value).present?
+               # if checked value is [1,2,4] and the intersection is [1,2], removes non-valid option 4
+               # and does final check to make sure it's not returning [], otherwise, defaults
+               selectbox_options & checked_value
+             elsif selectbox_options.include?(checked_value)
+               # checks if [1,2,3].includes?(3)
+               checked_value
+             else
+               default_value
+             end
+    {:refreshed_values => refreshed_values, :checked_value => @value, :read_only => read_only?, :visible => visible?}
+  end
+
+  def automate_output_value
+    return super unless force_multi_value
+    a = if @value.kind_of?(Integer)
+          [@value]
+        elsif @value.kind_of?(Array)
+          @value
+        else
+          @value.blank? ? [] : @value.chomp.split(',')
+        end
+    automate_values = a.first.kind_of?(Integer) ? a.map(&:to_i) : a
+    MiqAeEngine.create_automation_attribute_array_value(automate_values)
```

Comment:
@gmcculloug @d-m-u 
MiqAeEngine.create_automation_attribute_array_value([1,2,3]) => gives you back 1,2,3

For Automate to use it as an array the keyname should be prefixed with Array::

We have a helper method that does that
MiqAeEngine.create_automation_attribute_array_key("my_key")
which returns Array::my_key

You have to use the key and value together

In https://github.com/ManageIQ/manageiq/blob/master/app/models/dialog.rb#L50
We would have to detect that if it is a multi value field  it should use

result[MiqAeEngine.create_automation_attribute_array_key(df.automate_key_name)] = df.automate_output_value